### PR TITLE
Issue 250: Ground work to shift development to `main` from `develop`

### LINKED ISSUE #250: Drop use of `develop` branch
As the title we could drop the use of the `develop` branch approach we have been using now that we have R-Universe releases that are conditioned on GitHub releases and `pkgdown` docs that only switch to release mode when a non-development number is given to a package. 

This would have some nice benefits as it would make it clearer that development is happening and make sure people who want to are using the bleeding edge of the package (which we don't need to provide guarantees for) can easily. The downside is that it would potentially expose users, especially naive users who don't read the install instructions and just use standard `remote` tools, to the bleeding edge of the development version. It also makes it more likely we present a potentially broken package to some users though we obviously are trying not to do this via CI, reviews etc. 

If we wanted to implement this then we would need to: 

- [x] Update the README to remove mention of the `develop` branch, add a mention of release based installing and make it clear the preferred install path for a stable version is via our Universe. 
- [x] Remove mention of the `develop` branch from all contributing guides etc. 
- [x] After releasing `0.2.1` remove the `develop` branch. 
    *I've added this to the meta-release issue for `0.2.0`.*
- [x] If we are sure we don't want to revert then we may not to remove the special treatment of the `develop` branch in the GitHub Actions.
    *Planning on leaving this in as we may want to revert and it will break our current development setup.*
Tagging some people who might want to weigh in: @pearsonca, @Bisaloo, @adrian-lison, @epinowcast/developers, @sbfnk

Lets also have some 👍🏻 and 👎🏻 voting shall we 😄

### PR DESCRIPTION
**Pull request by seabbs**
## Description

This PR closes #250.

As discussed in #250 there are a range of benefits to shifting development to `main` with the key benefits of using `develop` having been eroded by enabling release-only building on R-universe. This PR updates the README and contributing guide to drop mentions of `develop` and make distinctions clear between the various available package versions. 

It also: 

- Adds an action to automatically render the `README.Rmd` when it is touched. 
- Fleshes out some of the install sections
- Adds a link out to our Docker container as an alternative.

This PR does not remove the `develop` triggers in our CI as this would break the current development workflow and does not remove the `develop` branch as we are still using it for this minor release.

## Checklist

- [x] My PR is based on a package issue and I have explicitly linked it.
- [x] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [x] I have read the [contribution guidelines](https://github.com/epinowcast/epinowcast/blob/main/CONTRIBUTING.md).
- [x] I have tested my changes locally.
- [x] I have updated the documentation if required.
- [x] My code follows the established coding standards.
- [x] I have added a news item linked to this PR.
- [x] I have updated the package development version by one increment in both `NEWS.md` and the `DESCRIPTION`.
- [x] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

**Review by Bisaloo — APPROVED** [review]

File: README.Rmd [hunk 1]
```diff
@@ -49,18 +48,20 @@ Install the latest released version of the package with:
...
 ```{r, eval = FALSE}
 remotes::install_github("epinowcast/epinowcast", dependencies = TRUE)
 ```
 
-The unstable development version can also be installed from GitHub using the following,
+Historical package releases can be installed from GitHub using `remotes` by adding the release tag as in the following example (which installs [`0.2.0`](https://github.com/epinowcast/epinowcast/releases/tag/v0.2.0)),
 
 ```{r, eval = FALSE}
-remotes::install_github("epinowcast/epinowcast@develop", dependencies = TRUE)
+remotes::install_github("epinowcast/epinowcast@v0.2.0", dependencies = TRUE)
```
Comment (pearsonca) [inline README.Rmd:60]: I think this version is better if people might want to conditionally use "HEAD" vs some other tag
```suggestion
remotes::install_github("epinowcast/epinowcast", dependencies = TRUE, ref = "v0.2.0")
```

File: NEWS.md [hunk 2]
```diff
@@ -40,6 +40,7 @@ This is release is in development. It is not yet ready for production use. If yo
 - Added an error message to `construct_rw()` when a random walk is specified for a variable that is not a numeric variable. See #228 by @seabbs and self-reviewed.
 - Added support for preprocessing and model fitting benchmarking using `touchstone` based on the implementation in `EpiNow2` by @sbfnk. See #200 by @seabbs, @adrian-lison, @sbfnk, and self-reviewed.
 - Added a complete set of data converters to map between line list (i.e. each row is a case) and count data (i.e incidence and cumulative counts by reference and report date). In particular, this will help workflows where individual line list data is available as it can now be formatted ready for preprocessing using a single call to `enw_linelist_to_incidence()` which previously took several steps. See #247 by @seabbs and @jhellewell14 and reviewed by @pearsonca.
+- Dropped the use of the `develop` branch for development versions of the package. This change was discussed in #250 with the major motivator being that since the introduction of release only builds to R Universe we no longer need to have a stable `main` branch of GitHub to control our releases. See #256 by @seabbs and reviewed by
```
Comment (pearsonca) [inline NEWS.md:43]: Well, the motivator is this approach is a stumbling block, and the benefits have gone away with other process changes. S'ok to say that approach had downsides and we determined they outweigh (now, not initially) the benefits.

File: README.Rmd [hunk 3]
```diff
@@ -33,11 +33,10 @@ Tools to enable flexible and efficient hierarchical nowcasting of right-truncate
...
 - [Directory of example scripts](https://github.com/epinowcast/epinowcast/tree/main/inst/examples): Not as fleshed out as our complete case studies these scripts are used during package development and each showcase a subset of package functionality. Often newly introduced features will be explored here before surfacing in other areas of our documentation.
-- [Community forum](https://community.epinowcast.org/): Our community forum is where development of tools is discussed, along with related research from our members and discussions between users. If you are interested in real-time analysis of infectious disease this is likely a good place to start regardless of if you end up making use of `epinowcast`.
+- [Community forum](https://community.epinowcast.org/): Our community forum is where development of methods and tools is discussed, along with related research from our members and discussions between users. If you are interested in real-time analysis of infectious disease this is likely a good place to start regardless of if you end up making use of `epinowcast`.
```
Comment (pearsonca) [inline README.Rmd:39]: What's the plan to for use of the github discussion board? I'm generally pro one-job-one-tool -- does it make sense to transition that board to something here? Or are there specific capabilities over that not on offer here?

File: README.md [hunk 4]
```diff
@@ -36,29 +36,29 @@ particular the following installation and quick start sections. As you
...
-    interested in real-time analysis of infectious disease this is
-    likely a good place to start regardless of if you end up making use
-    of `epinowcast`.
+Below is a short list of current resources. - [Package
+website](https://package.epinowcast.org/): This includes a function
+reference, model outline, and case studies making use of the package.
+This site refers to the release version of our package which can be
+installed from our Universe or from the latest GitHub release (see
+installation instructions). The development version of our documentation
+(corresponding to our `main` branch on GitHub) is available
```
Comment (pearsonca) [inline README.md:45]: Something mangled with generation here.

**Review by pearsonca — CHANGES_REQUESTED** [review]
Seems fine except: something to fix with the renderer?

A few other asides in comments; take 'em or leave 'em.

File: README.Rmd (hunk 3 above)
Comment (seabbs) [reply README.Rmd:39]: I thought we might want to use it for questions about how to use the package but not really sure.

**Pull request merged**